packages/auth/lib/AuthQueryHelper.ts

```diff
@@ -0,0 +1,200 @@
+import { DbGeneralPool, PgClient, PgPoolClient } from "@fullstack-one/db";
+import { ILogger } from "@fullstack-one/logger";
+import { CryptoFactory } from "./CryptoFactory";
+import { SignHelper } from "./SignHelper";
+
+export class AuthQueryHelper {
+  private dbGeneralPool: DbGeneralPool;
+  private logger: ILogger;
+  private authConfig: any;
+  private cryptoFactory: CryptoFactory;
+  private possibleTransactionIsolationLevels: [string, string, string, string] = [
+    "SERIALIZABLE",
+    "REPEATABLE READ",
+    "READ COMMITTED",
+    "READ UNCOMMITTED"
+  ];
+  private signHelper: SignHelper;
+
+  constructor(dbGeneralPool: DbGeneralPool, logger: ILogger, authConfig: any, cryptoFactory: CryptoFactory, signHelper: SignHelper) {
+    this.dbGeneralPool = dbGeneralPool;
+    this.authConfig = authConfig;
+    this.logger = logger;
+
+    this.cryptoFactory = cryptoFactory;
+    this.signHelper = signHelper;
+  }
+
+  /* DB HELPER START */
+  public async createDbClientAdminTransaction(
+    dbClient: PgPoolClient,
+    isolationLevel: "SERIALIZABLE" | "REPEATABLE READ" | "READ COMMITTED" | "READ UNCOMMITTED" = "READ COMMITTED"
+  ): Promise<PgPoolClient> {
+    const isolationLevelIndex = this.possibleTransactionIsolationLevels.findIndex((item) => isolationLevel.toLowerCase() === item.toLowerCase());
+    const isolationLevelToUse = this.possibleTransactionIsolationLevels[isolationLevelIndex];
+
+    // Begin transaction
+    await dbClient.query(`BEGIN TRANSACTION ISOLATION LEVEL ${isolationLevelToUse};`);
+    // set user (admin) for dbClient
+    await this.setAdmin(dbClient);
+    return dbClient;
+  }
+
+  public async createDbClientUserTransaction(
+    dbClient: PgPoolClient,
+    accessToken,
+    isolationLevel: "SERIALIZABLE" | "REPEATABLE READ" | "READ COMMITTED" | "READ UNCOMMITTED" = "READ COMMITTED"
+  ): Promise<PgPoolClient> {
+    const isolationLevelIndex = this.possibleTransactionIsolationLevels.findIndex((item) => isolationLevel.toLowerCase() === item.toLowerCase());
+    const isolationLevelToUse = this.possibleTransactionIsolationLevels[isolationLevelIndex];
+
+    // Begin transaction
+    await dbClient.query(`BEGIN TRANSACTION ISOLATION LEVEL ${isolationLevelToUse};`);
+    // set user for dbClient
+    await this.authenticateTransaction(dbClient, accessToken);
+    return dbClient;
+  }
+
+  public async getCurrentUserIdFromClient(dbClient) {
+    return (await dbClient.query("SELECT _auth.current_user_id();")).rows[0].current_user_id;
+  }
+
+  public async getCurrentUserIdFromAccessToken(accessToken) {
+    return this.userTransaction(accessToken, async (dbClient) => {
+      return this.getCurrentUserIdFromClient(dbClient);
+    });
+  }
+
+  // return admin transaction
```

Comment:
trivial comment

Reply:
Legacy code 😄

Reply:
I removed this and some more stupid comments in 0b57c7d49ec8afdcb312098aa4cd90e01b884dbe